### PR TITLE
Adds a code of conduct

### LINKED ISSUE #18: Code of Conduct
I believe that we should have a code of conduct. The [recommendation](https://opensource.guide/how-to-contribute/#anatomy-of-an-open-source-project) is to add a `CODE_OF_CONDUCT.md` file in each repository, although we could link to a document in this repository from that file. Having the file would allow us to complete the [Community Standards](https://github.com/nunit/nunit/community) checklist for each repository.

If we are planning or even just thinking of joining the .NET Foundation, we could just adopt their [code of conduct](https://dotnetfoundation.org/code-of-conduct).

We could also select one of GitHub's supplied [Code of Conduct Templates](https://help.github.com/articles/adding-a-code-of-conduct-to-your-project/). I haven't read through them yet, so I don't have an opinion on them.

### PR DESCRIPTION
**Pull request by rprouse**
Fixes #18 

I added my email as a point of contact for violations. I can change this if people prefer. I also added a short blurb to the README as recommended by [Contributor Covenant](http://contributor-covenant.org/). While updating the README, I noticed we had a todo to add the completed documents, so I did that at the same time.

I generated the CC using their `covgen` tool, so it was trivial to add. Based on that, we could easily add the same full CC to all projects. I am fine with just a `CODE_OF_CONDUCT.md` with just a link though. If we approve this CC, I think we should decide that at the same time.

I would also like to add the **Code of Conduct** blurb to each projects `README.md` and maybe a short blurb with a link to this repository?

**Do not merge, I will merge once we have consensus/vote**

### CODE REVIEW

File: README.md
Comment (jnm2) [inline README.md]: I'm not sure about the terminology of release. How about `Please note that all NUnit projects have a`?

**Review by jnm2 — CHANGES_REQUESTED** [review]

File: README.md
Comment (rprouse) [reply README.md]: This wording is the pluralized wording recommended at http://contributor-covenant.org/, but I am fine with the change. I will hold off on making the change until others comment.

Comment (jnm2) [reply README.md]: I didn't know they recommended that wording. In that case I have no opinion either way besides a small typo fix.

Comment (rprouse) [reply README.md]: I can't even pluralize correctly without spelling it wrong :smile: I will fix.

**Review by jnm2 — APPROVED** [review]

**Comment by jnm2** [comment]
> I would also like to add the Code of Conduct blurb to each projects README.md and maybe a short blurb with a link to this repository?

I like this strategy.

**Review by ChrisMaddock — APPROVED** [review]

**Comment by CharliePoole** [comment]
This looks good to me

**Comment by rprouse** [comment]
We now have 4/5 approval, I am going to merge.

**Pull request merged**
